src/helper/armis.py

```diff
@@ -0,0 +1,47 @@
+from helper.config import get_config_from_file
+from helper.filesystem import get_config_directory
+from armis import ArmisCloud
+from .database import MacList
+import re
+
+global_acloud = None
+armis_config = get_config_from_file(get_config_directory() / 'armis.cfg')
+
+
+def get_or_create_armis_cloud():  # connects to armis cloud
+    global global_acloud
+    if global_acloud is None:
+        global_acloud = ArmisCloud(
+            api_secret_key=armis_config['armis-server']['api_secret_key'],
+            tenant_hostname=armis_config['armis-server']['tenant_hostname']
+        )
+    return global_acloud
+
+
+def _filter_sort_sites(sites):  # removes all sites that dont match the regex-Pattern
+    pattern = rf"{armis_config['armis-server']['sites_pattern']}"
+    filtered_sites = {key: value for key, value in sites.items() if re.match(pattern, value['name'])}
+    return dict(sorted(filtered_sites.items(), key=lambda x: x[1]['name']))
+
+
+def get_armis_sites():
+    acloud = get_or_create_armis_cloud()
```

Comment:
The functions would be less complex, if "get_or_reate_cloud" would be implemented as Decorator. So all functions that need an armis connection would just use a "@armiscloud" decorator.
more infos on decorators can be found here: https://book.pythontips.com/en/latest/decorators.html